Scripts/Runtime/Data/VirtualData/VirtualDataExtensions.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// Helper methods when working with <see cref="VirtualData{TKey,TInstance}"/>
+    /// These make it more clear what is happening when operating on <see cref="VirtualData{TKey,TInstance}"/> instances
+    /// in a job.
+    /// </summary>
+    public static class VirtualDataExtensions
+    {
+        /// <summary>
+        /// Writes a result struct to the <see cref="VDJobResultsDestination{TResult}"/> on a
+        /// <see cref="IVirtualDataInstance{TResult}"/>
+        /// </summary>
+        /// <param name="instance">The instance to correspond the result to</param>
+        /// <param name="result">The result struct to write</param>
+        /// <param name="jobUpdater">The <see cref="VDJobUpdater{TKey,TInstance}"/> the instance was from.</param>
+        /// <typeparam name="TKey">The type of key for the instance.</typeparam>
+        /// <typeparam name="TInstance">The type of the instance struct.</typeparam>
+        /// <typeparam name="TResult">The type of the result struct</typeparam>
+        public static void Complete<TKey, TInstance, TResult>(this TInstance instance, ref TResult result, ref VDJobUpdater<TKey, TInstance> jobUpdater)
+            where TKey : struct, IEquatable<TKey>
+            where TInstance : struct, ILookupData<TKey>, IVirtualDataInstance<TResult>
+            where TResult : struct
+        {
+            VDJobResultsWriter<TResult> resultsWriter = instance.ResultsDestination;
+            resultsWriter.Add(ref result, jobUpdater.LaneIndex);
+            jobUpdater.Complete();
```

Comment:
Now that I see this in context I wonder if we should come up with a different name than `Complete()` for the method. Particularly since the word job is in there it makes me think this is a sync point.

Reply:
Let's chat about it. I agree the naming has been the hardest part. 

Here's the context for coming up with the name - see `ContinueOn` and `Complete`

```
            public void Execute(int index)
            {
                TimerInstance timerInstance = m_Timers[index];
                timerInstance.RemainingTime -= m_DeltaTime;
                if (timerInstance.RemainingTime > 0.0f)
                {
                    timerInstance.ContinueOn(ref m_Timers);
                    return;
                }

                timerInstance.Complete(new TimerResult(timerInstance.Key), ref m_Timers);
            }
```

Reply:
Some naming ideas:
 - Resolve (stolen from Promises)
 - Conclude
 - Finalize (probably not good either given existing meanings)
 - Fulfill
 - Done
 - FInish

Alternatively if we want to change continue you could come up with a different analogy:
Instead of ContinueOn/Complete
 - RetainWith/Release
 - PersistOn/Release
 - ScheduleUpdateWith/Conclude

I think my fave is just changing Complete to Resolve or doing Retain/Release.

Reply:
Cool, changing it to `Resolve` and keeping `Continue`